### PR TITLE
Jetpack Social: Disable Media Picker in sidebar when Social Image Generator is active

### PR DESCRIPTION
**Pull request by danielpost**
With this change we disable the media picker if SIG is enabled, to handle some confusion while we are not implementing the new UI.

More here: pdrWKz-TB-p2

## Proposed changes:

* Updated media section so it accepts a notice and a disabled prop
* Set the disabled prop based on SIG availability.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pdrWKz-TB-p2
## Does this pull request change what data or activity we track or use?

No.
## Testing instructions:

* Open the editor with the Advanced plan.
* You should no be able to pick/remove/interact with the media picker if SIG is enabled. Turning SIG off should re-enable the picker.

https://user-images.githubusercontent.com/36671565/234886786-35759da8-678e-4c01-a967-b1514609a0b3.mp4

### CODE REVIEW

File: projects/js-packages/publicize-components/src/components/form/index.js [hunk 1]
```diff
@@ -188,7 +190,19 @@ export default function PublicizeForm( {
 							/>
 						</>
 					) }
-					{ isEnhancedPublishingEnabled && <MediaSection /> }
+					{ isEnhancedPublishingEnabled && (
+						<MediaSection
+							disabled={ isImageGeneratorEnabled }
```
Comment (pablinos) [inline projects/js-packages/publicize-components/src/components/form/index.js]: This is definitely an odd state, but I published a post with SIG enabled and then turned the feature off. That meant that `isImageGeneratorEnabled` here is true, but the panel is hidden as isSocialImageGeneratorEnabled from `usePublicizeConfig` is false.

Do you think we should try and use both of these values? It seems like useImageGeneratorConfig shouldn't come back as enabled if the feature isn't available, although that could mean people lose the OG images if they edit a post after letting the Advanced plan lapse.

image

**Review by pablinos — COMMENTED** [review]
I think this is probably fine as is, but I just wondered about the 'enabled' flags.

File: projects/js-packages/publicize-components/src/components/form/index.js (hunk 1 above)
Comment (pablinos) [reply projects/js-packages/publicize-components/src/components/form/index.js]: I've added this change in [1996bbd](https://github.com/Automattic/jetpack/pull/30311/commits/1996bbd0840814208c86e4640d14fc9f7ada949b), but it's possibly not a situation that will ever occur!

File: projects/js-packages/publicize-components/src/components/form/index.js [hunk 2]
```diff
@@ -188,7 +192,19 @@ export default function PublicizeForm( {
 							/>
 						</>
 					) }
-					{ isEnhancedPublishingEnabled && <MediaSection /> }
+					{ isEnhancedPublishingEnabled && (
+						<MediaSection
+							disabled={ isSocialImageGeneratorEnabled && isImageGeneratorEnabled }
```
Comment (spsiddarthan) [inline projects/js-packages/publicize-components/src/components/form/index.js]: Super minor nitpick and we can address it later, `isImageGeneratorEnabled` should perhaps be renamed to `isImageGeneratorSelected`, the word 'enabled' makes me think if the feature is enabled or not.

**Review by spsiddarthan — DISMISSED** [review]
The feature works as intended. Just added a minor comment. I had to enable SIG from the admin page to test this. If the default state of SIG is disabled, without the admin screen in Jetpack, we can't really offer it to users on Jetpack, but we know that already, don't we?

File: projects/js-packages/publicize-components/src/components/form/index.js (hunk 2 above)
Comment (pablinos) [reply projects/js-packages/publicize-components/src/components/form/index.js]: Yeah I was a bit concerned about these names too. Perhaps `isImageGeneratorEnabledForPost`?

**Review by pablinos — APPROVED** [review]
LGTM!

**Comment by zinigor** [comment]
Cherry picked in 339299f14113b7c929f93030672328251acbe9c9.

**Pull request merged**